lib/Core/TerriaError.d.ts

```diff
@@ -0,0 +1,10 @@
+export = TerriaError;
+
+declare class TerriaError {
```

Comment:
Would it be reasonable to move `TerriaError` over to TypeScript instead of writing a separate declaration file for it?

Reply:
Yes. Can do that as another PR.

Reply:
Removed the ambient typing for now.